src/bindgen/ir/constant.rs

```diff
@@ -151,7 +151,25 @@ impl Literal {
             syn::Expr::Lit(syn::ExprLit {
                 lit: syn::Lit::Char(ref value),
                 ..
-            }) => Ok(Literal::Expr(format!("'{}'", value.value()))),
+            }) => {
+                let v = value.value();
+                if v.len_utf8() == 1 {
+                    Ok(Literal::Expr(format!(
+                        "'{}'",
+                        value.value().escape_default()
+                    )))
+                } else {
+                    Ok(Literal::Expr(format!(
+                        "L'{}'",
+                        value
```

Comment:
As mentioned earlier, it would be easier to avoid `escape_default` for this case altogether and just format manually as a hexadecimal integer with something like:

```
format!(r"L'\u{:X}'", value.value() as u32)
```